Make the core logic deal with tok instead of tez

gkaracha opened the pull request:
Though this transition is pretty menial, it is a necessary first step for addressing #213, independently of the path we end up choosing (conditional compilation vs. having a tez/FA2 wrapper contract). By making all the core logic deal with an opaque-ish type `tok` we can defer the choice of collateral type to a higher level. With this PR `checker.ml` deals with tez—as it did before—and makes calls to `tok_of_tez` before calling core logic and `tez_of_tok` when interpreting results from core logic. No other module makes calls to these conversion functions (excluding the tests ofc).

While working on this I also noticed that our API uses `tez` in names which we probably want to rename to something more generic: `Deposit_tez` and `Withdraw_tez` (which I think we could rename to `Deposit_collateral` and `Withdraw_collateral`).

utdemir reviewed (approved):
LGTM :). I fixed a trivial issue and rebased, so if you want to make changes you might need a hard reset.

gkaracha commented:
> LGTM :). I fixed a trivial issue and rebased, so if you want to make changes you might need a hard reset.

Thanks for having a look! :pray: No more changes I think, this PR is large enough as it is; I'll go ahead and merge it :slightly_smiling_face:
